basis/compiler/tree/propagation/simple/simple.factor

```diff
@@ -6,9 +6,11 @@ combinators combinators.short-circuit compiler.tree
 compiler.tree.propagation.constraints compiler.tree.propagation.info
 compiler.tree.propagation.inlining compiler.tree.propagation.nodes
 compiler.tree.propagation.slots continuations fry kernel make
-sequences sets stack-checker.dependencies words ;
+namespaces sequences sets stack-checker.dependencies words ;
 IN: compiler.tree.propagation.simple
 
+SYMBOL: word-being-compiled
```

Comment:
Is this information not available anywhere else?

Reply:
I also asked myself the same question and tried to trace in which context a word is being compiled.  It seems that definition is separated from compilation quite well.  

I set a breakpoint on `compile-word` and inspected the dynamic variables, but could not find any which would indicate which word is being compiled.

The alternative would be to change the interface of `frontend` to either take the word currently defined or return the return value info as additional stack output.

Even if I didn't miss something there, I don't know which version would be better.  Having the word being compiled in context during compilation and not touching the interface of `frontend` made the most sense to me.  Open to suggestions on this...